packages/@react-stately/data/test/useTreeData.test.js

```diff
@@ -540,4 +540,36 @@ describe('useTreeData', function () {
     expect(result.current.items[0].children[1].children[0]).toBe(initialResult.items[0].children[1].children[0]);
     expect(result.current.items[0].children[2]).toBe(initialResult.items[0].children[2]);
   });
+
+  it('should move an item to a new index within its current parent', function () {
```

Comment:
do we need a test that confirms nothing weird happens to the children since we are now touching them on moves?